From e1c519b3edbdf5b419f6e69ef1fd8c94227725cf Mon Sep 17 00:00:00 2001
From: Dirk Chen <dirkchen@amd.com>
Date: Tue, 16 Jun 2026 17:12:58 +0800
Subject: [PATCH] psusensor: skip sensor reads during firmware updates

This patch is based on the following PR [1].
A new MPS VR chip, the MPQ87xx series, has been introduced.
Add MPQ87XXFirmware to the lookup table.

[1] https://gerrit.openbmc.org/c/openbmc/dbus-sensors/+/83968

Signed-off-by: Dirk Chen <dirkchen@amd.com>
---
 src/psu/PSUSensorMain.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/psu/PSUSensorMain.cpp b/src/psu/PSUSensorMain.cpp
index 2b8b169..3df3761 100644
--- a/src/psu/PSUSensorMain.cpp
+++ b/src/psu/PSUSensorMain.cpp
@@ -195,6 +195,7 @@ static constexpr auto firmwareTypes = std::to_array<std::string_view>({
     "MP297XFirmware",
     "MP5998Firmware",
     "MP994XFirmware",
+    "MPQ87XXFirmware",
 });
 
 static boost::container::flat_map<std::string, std::shared_ptr<PSUSensor>>
-- 
2.53.0-Meta

